src/containers/Petition.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { fetchPetition } from 'actions/PetitionActions';
+import Petition from 'components/Petition';
+
+const PetitionContainer = React.createClass({
+  // When the component gets added to the DOM,
+  // fetch Petition if `id` is not defined.
+  componentDidMount () {
+    if (!this.props.id) {
+      this.props.fetchPetition(this.props.params.id);
+    }
+  },
+
+  render () {
+    return (
+      <div>
+        <Helmet title={this.props.title} />
+        <Petition {...this.props} />
+      </div>
+    );
+  }
+});
+
+PetitionContainer.fetchData = ({ store, params }) => {
+  return store.dispatch(fetchPetition(params.id));
+};
```

Comment:
This is the key for of server-side redux rendering. In `server/render.js` we grab this function and dispatch the `fetchPetition` action, to hydrate the store when the route hits. The page isn't rendered until the promise is fulfilled.